src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateDao.java

```diff
@@ -118,18 +120,24 @@
     @SqlUpdate("UPDATE mandates m SET state = :state WHERE m.id = :id")
     int updateState(@Bind("id") Long id, @Bind("state") MandateState mandateState);
 
-    @SqlUpdate("UPDATE mandates m SET state = :state FROM gateway_accounts g WHERE m.payment_provider_id = :providerMandateId " +
+    @SqlUpdate("UPDATE mandates m SET state = :state, state_details = :stateDetails, state_details_description = :stateDetailsDescription " +
+            "FROM gateway_accounts g WHERE m.payment_provider_id = :providerMandateId " +
             "AND g.id = m.gateway_account_id AND g.organisation = :goCardlessOrganisationId AND g.payment_provider = :provider")
     int updateStateByProviderIdAndOrganisationId(@Bind("provider") PaymentProvider paymentProvider,
                                                  @Bind("goCardlessOrganisationId") GoCardlessOrganisationId goCardlessOrganisationId,
                                                  @Bind("providerMandateId") PaymentProviderMandateId paymentProviderMandateId,
-                                                 @Bind("state") MandateState mandateState);
+                                                 @Bind("state") MandateState mandateState,
+                                                 @Bind("stateDetails") String details,
+                                                 @Bind("stateDetailsDescription") String detailsDescription);
 
-    @SqlUpdate("UPDATE mandates m SET state = :state FROM gateway_accounts g WHERE m.payment_provider_id = :providerMandateId " +
+    @SqlUpdate("UPDATE mandates m SET state = :state, state_details = :stateDetails, state_details_description = :stateDetailsDescription " +
+            "FROM gateway_accounts g WHERE m.payment_provider_id = :providerMandateId " +
             "AND g.id = m.gateway_account_id AND g.organisation IS NULL AND g.payment_provider = :provider")
     int updateStateByProviderId(@Bind("provider") PaymentProvider paymentProvider,
                                 @Bind("providerMandateId") PaymentProviderMandateId paymentProviderMandateId,
-                                @Bind("state") MandateState mandateState);
+                                @Bind("state") MandateState mandateState,
```

Comment:
Would be nice if its possible to `@BindBean` on the new `DirectDebitStateWithDetails` class to reduce the number of args and avoid deconstructing the contents of `DirectDebitStateWithDetails` prematurely...

Reply:
Hmm, didn’t think of that! Might change it later.